bpf/custom/README.rst

```diff
@@ -0,0 +1,125 @@
+..  Copyright (C) 2021 Authors of Cilium
+
+===================================
+Tail Call Hooks for Custom Programs
+===================================
+
+The files in this directory implement a sample byte counter program which can
+be attached to dedicated hooks in the datapath.
+
+.. note::
+    This is a beta feature. Please provide feedback and file a GitHub issue if
+    you experience any problems.
+
+Datapath Hooks
+==============
+
+Implementation
+--------------
+
+When compiled with ``ENABLE_CUSTOM_CALLS`` set, Cilium's datapath contains some
+tail call hooks permitting to insert custom programs in the datapath. These
+custom programs should be particularly careful *not* to interfere with the
+logics of the datapath, at the risk of altering or breaking the network
+connectivity in the cluster.
+
+Before the tail call happens, the datapath encodes two elements into the ``cb``
+field of the socket buffer for the packet:
+
+- The return value for that packet, so that the custom program can return it on
+  exit. This is because a tail call is not like a function call, and the custom
+  program does not *return* into the core datapath logics. Therefore, the
+  custom program must be called as the very last step of the processing of the
+  packet, and must preserve the return value selected by the datapath.
+
+- The identity corresponding to the packet, that is, the Cilium identity
+  associated to the sender for the ingress hook, and to the receiver when on
+  the egress hook. This identity is not strictly necessary to the custom
+  program, but it prove useful for some use cases. Given that the datapath
+  already did the work for retrieving the identity for the packet, it just
+  passes it down to the custom programs, in case it saves them the trouble to
+  extract it a second time.
+
+Limitations
+-----------
+
+- There is currently no hook on the egress path for socket-based
+  load-balancing.
+
+Sample Byte Counter Program
+============================
+
+This directory contains a sample byte counter program, to help count how many
```

Comment:
```suggestion
This directory contains a sample byte-counter program, to help count how many
```

Reply:
Updated elsewhere in this doc and CI test.